test/ConsoleArgumentParser.test.cpp

```diff
@@ -38,20 +36,3 @@ TEST(ConsoleArgumentParser, NoSwitchArgument) {
 TEST(ConsoleArgumentParser, TooManyArguments) {
 	EXPECT_THROW(FindModuleDirectory(std::vector<std::string> { "/loadmod", "path1", "path2" }), std::runtime_error);
 }
-
-
-// Returns true if calling FindModuleDirectory provides input to the logger by checking log size
-::testing::AssertionResult FindModuleDirectoryIsLogged(
```

Comment:
I wanted to see this morph into looking for strings on the last logged line at some point to better test PostErrorMessages. I suppose it can be re-added in the future if we implement it in that form.

Reply:
I was thinking about that sort of thing too. This might be a good problem to try Google Mock on. If we can create a mock object for the logger, than we can verify more directly if we are logging proper error messages, without needing to actually go to disk.

To test with mock objects (with or without Google Mock), we'd need to make a few changes to the logging code. We'd have to split the `Logger` class into an interface class, and a separate implementation. This might be a good idea to do anyway, since it would allow for multiple logging output options.

We would also need to change the global `logger` variable to be a pointer or reference to a `Logger`. That would allow method calls to go through to a potential derived class. I'd recommend a pointer if we want to allow for updates to the logging output at runtime, allowing it to be swapped out for a different type of logger. A global reference can only be set once, and it would be done during global initialization.

Not required, but something I was thinking about. Perhaps we should use global logging functions, rather than a global logging object. The global functions can then delegate to the logging object internally. That would allow restricting access to the logging object, and possibly making it not a global. That could have benefits for deferring the creation of the logging object. It would also protect the code from having to change `logger.` to `logger->` all over the place if we change it from a `Logger` instance to a pointer.

Finally, with Google Mock / Google Test support, we might potentially use the string matchers on the error messages. That would allow for testing of keywords, rather than exact error messages. Small changes in the error messages wouldn't immediately break tests, and we also wouldn't have to export the error strings to give tests access. As an example, suppose a file API has an error accessing a file. We might check that the error message contains the filename. Or we might check that the error message contains the path and filename if the directory portion is likely to be relevant.

As a final thought, if we want to allow for parallel unit tests on multiple threads, the (global) `logger` variable could be made into a thread local variable. That way each unit test, running on it's own thread, could have it's own mock logger object, allowing the unit tests to run in parallel without stepping on each other. This might be a good idea anyway, as logging from multiple threads would either need separate loggers, or the logger would have to be multithread aware and provide for proper safety or synchronization.

So yes, we should morph things into better tests for decent error messages. :)